File: leetcode/1.two-sum.go
/*
 * @lc app=leetcode id=1 lang=golang
 *
 * [1] Two Sum
 */

// @lc code=start
func twoSum(nums []int, target int) []int {
    m := make(map[int]int)
	for i, n := range nums {
		complement := target - n
		if val, ok := (m[complement]); ok {
			return []int{val, i}
		}
		m[n] = i
	}
	return nil
}
// @lc code=end

File: leetcode/1.two-sum.js
/*
 * @lc app=leetcode id=1 lang=javascript
 *
 * [1] Two Sum
 */

// @lc code=start
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    const map = new Map();

    for (let i = 0; i < nums.length; i++) {
       const complement = target - nums[i];
       if (map.has(complement)) {
          return [map.get(complement), i];
       }
       map.set(nums[i], i);
    }
    return []
};
// @lc code=end

